helm/policies-azure/tests/abs/test_azure_default.py

```diff
@@ -0,0 +1,29 @@
+import yaml
+from functools import partial
+import time
+import random
+import string
+import ensure
+from textwrap import dedent
+
+from ensure import release
+from ensure import cluster
+from ensure import azurecluster
+
+import pytest
+from pytest_kube import forward_requests, wait_for_rollout, app_template
+
+import logging
+LOGGER = logging.getLogger(__name__)
+
+
+@pytest.mark.smoke
+def test_azure_cluster_policy(release, cluster, azurecluster) -> None:
+    """
+    test_azure_cluster_policy tests defaulting of an AzureCluster where all required values are empty strings.
+
+    :param release: Release CR which is used by the Cluster.
+    :param cluster: Cluster CR which uses the release and matches the AzureCluster.
+    :param azurecluster: AzureCluster CR with empty strings which matches the Cluster CR.
+    """
+    assert azurecluster['metadata']['labels']['cluster.x-k8s.io/watch-filter'] == ensure.capa_version
```

Comment:
It looks like `release` and `cluster` aren't used -- does the function signature just have to include those for interface reasons?

Reply:
It's the magic of python fixtures :sparkles: 
The arguments mean that the fixture will create the objects and clean them up later again - so during the testcase a release and cluster CR is created and deleted afterwards. It doesn't matter if the actual test accesses it. See it as "requesting" those resources.

Reply:
![](https://cdn.shopify.com/s/files/1/0264/0075/1725/t/7/assets/pf-e8e78f71--Magic-Tricks.gif?v=1620253927)